.github/workflows/ci.yml

```diff
@@ -35,7 +35,9 @@ jobs:
       MAX_COUNT: 6
     steps:
       - name: Determine fetch depth
-        run: echo "FETCH_DEPTH=$((${{ github.event.pull_request.commits }} + 2))" >> "$GITHUB_ENV"
+        # Fetch some extra commits for pull requests on forks that don't build
+        # on a merge commit.
+        run: echo "FETCH_DEPTH=$((${{ github.event.pull_request.commits }} + 100))" >> "$GITHUB_ENV"
```

Comment:
In commit "ci: increase fetch depth for 'test each commit'" (6398efbe145fb297957060600615262d59f2b47c)

I think the following would be a more direct fix. I didn't test it but it just omits the `^${MERGE_BASE}^@` argument when there is no merge base, which should be fine because there should be no need to exclude merge base ancestors in that case:

```diff
-          echo "TEST_BASE=$(git rev-list -n$((${{ env.MAX_COUNT }} + 1)) --reverse HEAD ^$(git rev-list -n1 --merges HEAD)^@ | head -1)" >> "$GITHUB_ENV"
+          MERGE_BASE=$(git rev-list -n1 --merges HEAD)
+          EXCLUDE_MERGE_BASE_ANCESTORS=
+          if test -n "$MERGE_BASE"; then
+              EXCLUDE_MERGE_BASE_ANCESTORS=^${MERGE_BASE}^@
+          fi
+          echo "TEST_BASE=$(git rev-list -n$((${{ env.MAX_COUNT }} + 1)) --reverse HEAD $EXCLUDE_MERGE_BASE_ANCESTORS | head -1)" >> "$GITHUB_ENV"
```

Reply:
Thanks, will try.